.github/workflows/autoyapf.yml

```diff
@@ -1,8 +1,33 @@
 name: Autoyapf
-
-on: [push, pull_request]
+on:
+  push:
+    paths:
+    - '**.py'
 jobs:
   autoyapf:
     runs-on: ubuntu-latest
     steps:
-      - run: echo "In the future this will run the automatic formatting"
+      - uses: actions/checkout@master
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Install dependencies
+        run: |
+          sudo apt-get -y install libbz2-dev
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          pip install yapf toml
+      - name: autoyapf
+        id: autoyapf
+        run: |
+          yapf --style google --recursive --in-place .
+      - name: Check for modified files
+        id: git-check
+        run: echo "modified=$(if git diff-index --quiet HEAD --; then echo "false"; else echo "true"; fi)" >> $GITHUB_ENV
```

Comment:
I'm wondering if it might be better to just fail (instead of creating commits we are unaware of). I'm thinking of the case where yapft or pylint update in a way yapft creates problems for pylint and thus we end up in a test-failing bug without being aware